src/pipeline_factory.cpp

```diff
@@ -272,6 +272,60 @@ Status PipelineDefinition::validateNodes(ModelManager& manager) {
     return StatusCode::OK;
 }
 
+Status PipelineDefinition::getInputsInfo(tensor_map_t& inputsInfo, ModelManager& manager) const {
+    // Assumptions: this can only be called on available pipeline definition.
+    // Add check if available when pipeline status will be implemented.
+
+    static const auto byName = [](const std::string& name) {
+        return [name](const NodeInfo& nodeInfo) {
+            return nodeInfo.nodeName == name;
+        };
+    };
+
+    for (const auto& [dependant_node_name, all_mappings] : connections) {
+        const auto& dependant_node_info = std::find_if(std::begin(nodeInfos), std::end(nodeInfos), byName(dependant_node_name));
+        for (const auto& [dependency_node_name, specific_dependency_mapping] : all_mappings) {
+            const auto& dependency_node_info = std::find_if(std::begin(nodeInfos), std::end(nodeInfos), byName(dependency_node_name));
+            if (dependency_node_info->kind != NodeKind::ENTRY) {
+                continue;
+            }
+
+            switch (dependant_node_info->kind) {
+            case NodeKind::EXIT: {
+                for (const auto& [alias, real_name] : specific_dependency_mapping) {
+                    inputsInfo.insert({alias, TensorInfo::getUnspecifiedTensorInfo()});
+                }
+                break;
+            }
+            case NodeKind::DL: {
+                auto instance = manager.findModelInstance(dependant_node_info->modelName, dependant_node_info->modelVersion.value_or(0));
+                if (!instance) {
+                    SPDLOG_DEBUG("Model:{} was unavailable during pipeline:{} inputs info fetching", instance->getName(), this->getName());
```

Comment:
I am wondering if we shouldn't for now keep it at SPDLOG_INFO level during phase 2 development and at the end go through and switch to SPDLOG_DEBUG. Maybe add // TODO LOWER_LOG_LEVEL?
Especially when we will be doing concurrent tests debug.

Reply:
Done